app/test/vesting_test.go

```diff
@@ -0,0 +1,936 @@
+package app_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/celestiaorg/celestia-app/app"
+	"github.com/celestiaorg/celestia-app/app/encoding"
+	"github.com/celestiaorg/celestia-app/test/util/testfactory"
+	"github.com/celestiaorg/celestia-app/test/util/testnode"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	vestingtypes "github.com/cosmos/cosmos-sdk/x/auth/vesting/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	distributiontypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	tmtime "github.com/tendermint/tendermint/types/time"
+	"google.golang.org/grpc"
+)
+
+const (
+	totalAccountsPerType = 300
+	initBalanceForGasFee = 10
+)
+
+type _accountDispenser struct {
+	names   []string
+	counter int
+}
+
+type _accountType int
+
+const (
+	RegularAccountType _accountType = iota + 1
+	DelayedVestingAccountType
+	PeriodicVestingAccountType
+	ContinuousVestingAccountType
+)
+
+type VestingModuleTestSuite struct {
+	suite.Suite
+
+	accounts    sync.Map // map[_accountType]_accountDispenser
+	accountsMut sync.Mutex
+
+	kr   keyring.Keyring
+	cctx testnode.Context
+	ecfg encoding.Config
+}
+
+func TestVestingModule(t *testing.T) {
+	suite.Run(t, new(VestingModuleTestSuite))
```

Comment:
Agreed let's do it.
It takes some time because the vesting accounts are time sensitive and to assert some particular unlocking schemes, the test code waits for a number of seconds.